### PR TITLE
Adds party grenades

### PR DESCRIPTION
**Pull request by FlattestGuitar**
Adds party grenades that can spam confetti. Fun stuff.
They are currently two ways you can get them:
* they're a tier 2, 65 ticket reward in the prize counter
* there are two of those in the cargo party equipment crate
confetti!

Changed the confetti sprite, this is the new one.

:cl: FlattestGuitar
add: Adds confetti grenades.
/:cl:

### CODE REVIEW

**Comment by CrAzYPiLoT-SS13** [comment]
The confetti looks a bit...hefty. Maybe make them smaller? Other than that
:+1:

**Comment by Regen1** [comment]
confetti is way too big

**Comment by ProperPants** [comment]
@FlattestGuitar Make the confetti only take up 1/4th of a tile, then add four of those per tile. That would be a better size. Otherwise :+1:

**Comment by Fox-McCloud** [comment]
Sprites aren't very good, at all.

File: code/game/objects/items/weapons/grenades/confetti.dm [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	var/turf/T = get_turf(src)
+	playsound(T, 'sound/effects/confetti_partywhistle.ogg', 100, 1)
+	for(var/i=1, i<=20, i++) //20 confettis. Yes.
+		var/atom/movable/x = new spawner_type
+		x.loc = T
+		for(var/j = 1, j <= rand(1, 4), j++)
+			step(x, pick(NORTH,SOUTH,EAST,WEST))
+
+	qdel(src)
+	return
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/grenades/confetti.dm]: Trailing return. No biggie, but pointless

**Comment by NTSAM** [comment]
i like the sprites personally

**Comment by Spacemanspark** [comment]
I'd say it's fine as is, but maybe shrink it down a tad.

**Comment by FlattestGuitar** [comment]
@Spacemanspark, https://www.freesound.org/people/Marty2011/sounds/168855/

File: code/game/objects/items/weapons/grenades/confetti.dm [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	icon_state = "confetti"
+	var/spawner_type = /obj/effect/decal/cleanable/confetti
+
+/obj/item/weapon/grenade/confetti/prime()
+	var/turf/T = get_turf(src)
+	playsound(T, 'sound/effects/confetti_partywhistle.ogg', 100, 1)
+	for(var/i=1, i<=20, i++) //20 confettis. Yes.
+		var/atom/movable/x = new spawner_type
+		x.loc = T
+		for(var/j = 1, j <= rand(1, 4), j++)
```
Comment (Iamgoofball) [inline code/game/objects/items/weapons/grenades/confetti.dm]: not optimized

File: code/game/objects/items/weapons/grenades/confetti.dm [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/obj/item/weapon/grenade/confetti
+	name = "party grenade"
+	desc = "Party time!"
+	icon_state = "confetti"
+	var/spawner_type = /obj/effect/decal/cleanable/confetti
+
+/obj/item/weapon/grenade/confetti/prime()
+	var/turf/T = get_turf(src)
+	playsound(T, 'sound/effects/confetti_partywhistle.ogg', 100, 1)
+	for(var/i=1, i<=20, i++) //20 confettis. Yes.
```
Comment (Iamgoofball) [inline code/game/objects/items/weapons/grenades/confetti.dm]: not optimized

use
`for(var/i in 1 to somevalue)`

**Comment by FalseIncarnate** [comment]
Looks good. The confetti sprite is kinda big, but I'm willing to chalk that up to "there is a TON of confetti" and to make it easier for people to click to clean it. A potentially fun thing to do would be to make the confetti turn to ashes when burned with a fire or something, but that's not an important bit to have for initial merge.

**Comment by Spacemanspark** [comment]
That sound is perfect.

**Comment by FlattestGuitar** [comment]
Also, I forgot to mention - confetti can be cleaned with space cleaner and all the standard cleaning equipment in the janitor's arsenal.

**Pull request merged**